pandas/core/arrays/datetimes.py

```diff
@@ -896,7 +897,7 @@ def tz_convert(self, tz) -> Self:
 
         # No conversion since timestamps are all UTC to begin with
         dtype = tz_to_dtype(tz, unit=self.unit)
-        return self._simple_new(self._ndarray, dtype=dtype, freq=self.freq)
```

Comment:
I don't think this is an appropriate fix

Reply:
Why do you think so? and what would be an appropriate fix?

Reply:
Based on the failing tests seems like this behavior was relied upon

Reply:
Resolving this issue is undoing work done a while ago. The frequency attribute was set as part of this issue [#25241 (https://github.com/pandas-dev/pandas/issues/25241) - why do you think it should not be reset?

Reply:
Based on the failing unit test the freq should be preserved